store/root/store.go

```diff
@@ -337,6 +370,31 @@ func (s *Store) Prune(version uint64) error {
 	return nil
 }
 
+// StartMigration starts the migration process. It sets the migration manager
+// and initializes the channels. An error is returned if migration is already in
+// progress.
+// NOTE: This method should only be called once after loadVersion.
+func (s *Store) StartMigration() error {
+	if s.isMigrating {
+		return fmt.Errorf("migration already in progress")
+	}
+
+	// buffer the changeset channel to avoid blocking
```

Comment:
This comment is misleading, we can still block.

```suggestion
	// buffer at most 1 changeset, if the receiver is behind attempting to buffer more than 1 will block.
```